File: template-method/template.js
class Reporte {
    generar() {
        this.obtenerDatos();
        this.formatear();
    }

    obtenerDatos() {
        console.log("Obteniendo datos...");
    }

    formatear() {}
}

class ReportePDF extends Reporte {
    formatear() {
        console.log("Formato PDF");
    }
}

File: template-method/template.py
class Reporte:
    def generar(self):
        self.obtener_datos()
        self.formatear()

    def obtener_datos(self):
        print("Obteniendo datos...")

    def formatear(self):
        pass


class ReportePDF(Reporte):
    def formatear(self):
        print("Formato PDF")
